### PR TITLE
trans: unmask libusb errors

### PR DESCRIPTION
**Pull request by aleeraser**
When using uuu it can happen that libusb operations sometimes return with an error code, which is currently masked.

This prints the error code and propagates it.

### CODE REVIEW

File: libuuu/trans.cpp [hunk 1]
```diff
@@ -97,28 +97,35 @@ int TransBase::read(vector<uint8_t> &buff)
 int USBTrans::open(void *p)
 {
 	m_devhandle = p;
+	char err_buf[256];
+	int libusb_ret = 0;
```
Comment (nxpfrankli) [inline libuuu/trans.cpp:101]: string_ex err；

File: libuuu/trans.cpp [hunk 2]
```diff
@@ -97,28 +97,35 @@ int TransBase::read(vector<uint8_t> &buff)
...
-		int ret = libusb_detach_kernel_driver((libusb_device_handle *)m_devhandle, 0);
-		if(ret <0 && ret != LIBUSB_ERROR_NOT_SUPPORTED)
+		libusb_ret = libusb_detach_kernel_driver((libusb_device_handle *)m_devhandle, 0);
+
+		if(libusb_ret < 0 && libusb_ret != LIBUSB_ERROR_NOT_SUPPORTED)
 		{
-			set_last_err_string("detach kernel driver failure");
-			return -1;
+			sprintf(err_buf, "detach kernel driver failure (%d)", libusb_ret);
+			set_last_err_string(err_buf);
```
Comment (nxpfrankli) [inline libuuu/trans.cpp:111]: err.format("detach kernel driver failure (%d)", libusb_ret);

File: libuuu/trans.cpp [hunk 3]
```diff
@@ -188,10 +197,9 @@ int HIDTrans::write_simple(void *buff, size_t size)
 
 	if (ret < 0)
 	{
-		string err;
-		err = "HID(W):";
-		err += libusb_error_name(ret);
-		set_last_err_string(err);
+		char err_buf[100];
+		sprintf(err_buf, "HID(W): %s (%d)", libusb_error_name(ret), ret);
```
Comment (nxpfrankli) [inline libuuu/trans.cpp:201]: using string_ex

**Review by nxpfrankli — COMMENTED** [review]
using help class string_ex

**Comment by aleeraser** [comment]
> using help class string_ex

done 👍🏻

**Pull request merged**